config_management/domains/views.py

```diff
@@ -0,0 +1,151 @@
+        except:
+            return JsonResponse({'error_message':'Kindly post the feature id list under \'feature_id_list\' key'})
+
+        if len(feature_id_list)>0:
+            for feature_id in feature_id_list:
+                if type(feature_id) != str and type(feature_id) != int:
+                    return JsonResponse({'error_message':'post feature ids in integer/string format'})
+
+                try:
+                    feature_id_list = list(map(lambda a:int(a), feature_id_list))
+                except:
+                    return JsonResponse({'error_message':'post feature ids in integer/string format'})
+
+                try:#doubt on reducing database calls
+                    feature_object = Feature.objects.get(id = int(feature_id))
+                except:
+                    return JsonResponse({'error_message':'feature with this id does not exist'})
+
+        DomainFeatureMapping.objects.filter(domain=domain_object).all().delete()
+        
+        if len(feature_id_list)>0:
+            for feature_id in feature_id_list:
+                feature_object = Feature.objects.get(id = int(feature_id))
+                DomainFeatureMapping.objects.create(domain = domain_object, feature = feature_object)
+
+        domain_info = domain_object.get_info()
+        domain_info['feature_id_list'] = feature_id_list
+
+        args = {'domain_info':domain_info}
+        return JsonResponse(args)
+
+class DomainGetPost(View):
+
+    def get(self, request): #fetch list of all the domains
+
+        domains = Domain.objects.all() #fetch all domain objects
+        domains_info = []
+
+        for domain in domains:
+            domain_all_info = domain.get_info()
+            domains_info.append(domain_all_info) #storing name and id of all domains in a list
+
+        args = {'domains_info':domains_info}
+        return JsonResponse(args) #returning list containing domain's ids and names
+
+
+    def post(self, request):
+        
+        try: #raising error if body cannot be parsed 
+            json_data = json.loads(str(request.body, encoding='utf-8'))
+        except:
+            return JsonResponse({'error_message':'posted body cannot be parsed, post in json format'})
+
+        try: #raising error if domain name is not under 'name' key
+            domain_name = (json_data['domain_name'])
+        except:
+            return JsonResponse({'error_message':'Kindly post the domain name under \'domain_name\' key'})
+
+        try:
+            feature_id_list = (json_data['feature_id_list'])
+        except:
+            return JsonResponse({'error_message':'Kindly post the feature id list under \'feature_id_list\' key'})
+
+        if not validators.domain(domain_name):
+            return JsonResponse({'error_message':'Domain name is invalid, kindly enter a valid domain name'})
+
+        #checking if domain with provided name already exists
+        if Domain.objects.filter(name=(domain_name.lower())).exists(): 

```

Comment:
We can do lower to the domain_name at one place, don't repeat the code, have it re-usable.